### PR TITLE
Challenge "difference" - strategies list: Set/reduce/ filter/for loop

### LINKED ISSUE #1: group 1;difference: strategy tba

### LINKED ISSUE #2: Challenge "difference" strategies list: Set/ reduce/ filter/ for loop
Create and edit challenge "split object" file:

- [x] sandbox.js
- [x] difference.spec.js

### PR DESCRIPTION
**Pull request by AlinaTaoRao**
---
name: solution code review
about: A template PR for code review, complete with checklist
title: 'Code Review: challenge name, strategy name'
---
close #1 , #2

Before the reviewer merges this PR to main/master, all of these boxes must be
checked. Only the reviewer can check boxes.

## Behavior

> What the function does - inputs and outputs

- **The Function**
  - [x] The function's name is clear and helpful
  - [x] The parameter names are clear and helpful
  - [x] Default parameters are used if possible
- **JSDoc Comment**
  - [x] The docstring is well-formatted
  - [x] There are no spelling mistakes
  - [x] The description of the function's behavior is clear
  - [x] All parameters are documented with types
  - [x] Documented parameter names match the function's parameter names
  - [x] Parameters have a good description
  - [x] If default parameters are used, they are the correct type
  - [x] The return value is documented with a type
  - [x] The return value has a clear description
- **README**
  - [ ] All the sections in the solution README are complete
  - [ ] Descriptions in the README are thorough and clear
  - [ ] There is at least 2 use-cases
- **Unit Tests**
  - [x] Unit tests are present
  - [ ] There are at least 20 test cases
  - [ ] All `describe`s and `it`s have clear and helpful names
  - [ ] All test names actually describe what is being tested
  - [x] Tests are grouped logically using `describe`
  - [x] There is only one `expect` inside every `it`
  - [x] There is no logic (loops, conditionals, ...) in the test cases
  - [ ] Side-effects are tested, if necessary
  - [x] All tests pass in the CI check
  - [ ] The tests have 100% coverage of the solution
  - [ ] Tricky edge cases are tested (extra credit)

## Strategy

> The logic behind the solution

- **The Function**
  - [ ] Variables have names that help understand the strategy
  - [x] Spacing and formatting help to understand the strategy
  - [x] The code is written as simply and clearly as possible
  - [x] Use short, clear comments to describe the strategy
- **README**
  - [ ] There is an explanation of the solution's strategy in the README
  - [ ] The explanation is clear and understandable
  - [ ] The explanation actually matches the function that was written

## Implementation

> The code used to write the solution

- **Formatting**
  - [x] No formatting errors appear in the linting check
- **Linting**
  - [x] There should be no linting errors or warnings in the CI check
  - [x] Spelling should be correct in all code, comments and docstrings
- **The solution file**
  - [ ] Has a JSDoc comment
  - [ ] Has an exported function
  - [ ] Does no other code or comments outside of the function body (you can use
        `sandbox.js` for saving extra ideas)
- **The function body _Should_ ...**
  - [x] Be well and consistently formatted
  - [ ] Have clear and helpful variable names
  - [x] Declare variables in the smallest scope possible
  - [x] Be a simple and readable implementation
- **The function body should _Not_ ...**
  - [x] Have extra comments that are not about the solution's strategy
  - [x] Have any user interactions (`prompt`, `alert`, `confirm`)
  - [x] Have any `console` calls (`.log`, `.assert`, ...)
  - [x] Have any `debugger` statements

### CODE REVIEW

File: src/difference/alina/sandbox.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+ */
+
+// 1st Try: use Set //works
+/* strategy: 
+1) convert both arrays to sets that have no identical elements.
+2) get the set consisting of the elements that belong to the first array and that do not belong to the second array.
+3) convert the obtained set to a new array and return it.
+*/
+export const difference1 = (array = [], values = []) => {
+  const set1 = new Set(array);
```
Comment (aviv82) [inline src/difference/alina/sandbox.js]: change these variable names please

File: src/difference/solution-name.spec.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        const actual = solution(array);
+        expect(actual).toEqual([]);
+      });
+      it('both argument are empty', () => {
+        const actual = solution();
+        expect(actual).toEqual([]);
+      });
+    });
+
+    describe('standard test', () => {
```
Comment (aviv82) [inline src/difference/solution-name.spec.js]: test description should be more detailed. what are you testing?

example: array is an array with different data types

File: src/difference/solution-name.spec.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      it('number array', () => {
```
Comment (aviv82) [inline src/difference/solution-name.spec.js]: it case tests need to include the word 'should'

example: should return number array

**Review by aviv82 — CHANGES_REQUESTED** [review]
great job! looks good to me

[32-34] change 'set' variable names to arrayCopy, valuesCopy, newArray

maybe add more tests for edge cases.

maybe try to add guards, and tests for guards

File: src/difference/solution-name.spec.js (hunk 3 above)
Comment (AlinaTaoRao) [reply src/difference/solution-name.spec.js]: `should`... got it.

File: src/difference/solution-name.spec.js (hunk 2 above)
Comment (AlinaTaoRao) [reply src/difference/solution-name.spec.js]: detail added.

File: src/difference/alina/sandbox.js (hunk 1 above)
Comment (AlinaTaoRao) [reply src/difference/alina/sandbox.js]: Variable name changed. 😄

**Review by aviv82 — APPROVED** [review]
